src/qt/bitcoingui.cpp

```diff
@@ -1458,12 +1461,14 @@ void BitcoinGUI::configClicked()
 
 void BitcoinGUI::diagnosticsClicked()
 {
-#ifdef WIN32
-	if (!bGlobalcomInitialized) return;
-	qtSetSessionInfo(DefaultWalletAddress(), GlobalCPUMiningCPID.cpid, GlobalCPUMiningCPID.Magnitude);
-	bool result = PushGridcoinDiagnostics();
-    globalcom->dynamicCall("ShowDiagnostics()");
-#endif
+//#ifdef WIN32
+//	if (!bGlobalcomInitialized) return;
+//	qtSetSessionInfo(DefaultWalletAddress(), GlobalCPUMiningCPID.cpid, GlobalCPUMiningCPID.Magnitude);
+//	bool result = PushGridcoinDiagnostics();
+//    globalcom->dynamicCall("ShowDiagnostics()");
+//#endif
+    DiagnosticsDialog diag;
+    diag.exec();
```

Comment:
Open the window as non-modal? Like dialog.show? So the whole wallet does not freeze.
(i dont know how to do that)